src/olympia/files/utils.py

```diff
@@ -82,19 +82,23 @@ def get_filepath(fileorpath):
     return fileorpath
 
 
-def id_to_path(pk):
+def id_to_path(pk, depth=2):
     """
     Generate a path from an id, to distribute folders in the file system.
     1 => 1/1/1
     12 => 2/12/12
     123456 => 6/56/123456
+
+    depth can be specified through the corresponding parameter, defaults to 2,
+    levels deep, minimum is 1. depth=3 would return 6/56/456/123456 for 123456.
```

Comment:
With only 10 folders at each level we do end up with many sparsely populated interim folders, followed by a lot in the end level.  

For example, if we have 1 million end folders (add-on ids are trending upwards towards that): then for `depth=2` it would be 10 x 10 x 10000, and for `depth=3` we would have 10 x 10 x 10 x 1000.  

But if we group by 2 digits then its 123456 => 56/3456/123456 for `depth=2` (100 x 100 x 100) which is manageable at all the levels, and higher depths wouldn't be needed until we're beyond 10 million (so just fix it at `depth=2`).

Reply:
Yeah, I thought about that and then decided to stick to the original plan because it was easier, but I'm going to look into that. My main concern is around low ids - there is only one edge case with the implementation I went for, there would be a bit more with that approach, but maybe that's not a big deal.